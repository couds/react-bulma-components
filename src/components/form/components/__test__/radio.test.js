import React from 'react';
import renderer from 'react-test-renderer';
import Radio from '../radio';

describe('Radio component', () => {
  it('Should exist', () => {
    expect(Radio).toMatchSnapshot();
  });
  it('Should have radio classname', () => {
    const component = renderer.create(
      <Radio name="test">
        Test <a>Give me</a>
      </Radio>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Radio name="test" className="other-class this-is-a-test">
        <p>Default</p>
      </Radio>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(
      <Radio name="test" style={{ height: 250 }}>
        <p>Default</p>
      </Radio>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be disabled, checked and with value', () => {
    const component = renderer.create(
      <Radio name="test" value="TEST" checked disabled>
        <p>Default</p>
      </Radio>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
