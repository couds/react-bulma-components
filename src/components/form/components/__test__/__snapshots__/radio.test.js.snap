// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component Should be disabled, checked and with value 1`] = `
<label
  className="radio"
  disabled={true}
>
  <input
    checked={true}
    disabled={true}
    name="test"
    type="radio"
    value="TEST"
  />
  <p>
    Default
  </p>
</label>
`;

exports[`Radio component Should concat classname in props with Bulma classname 1`] = `
<label
  className="radio other-class this-is-a-test"
  disabled={false}
>
  <input
    checked={false}
    disabled={false}
    name="test"
    type="radio"
    value=""
  />
  <p>
    Default
  </p>
</label>
`;

exports[`Radio component Should exist 1`] = `[Function]`;

exports[`Radio component Should have radio classname 1`] = `
<label
  className="radio"
  disabled={false}
>
  <input
    checked={false}
    disabled={false}
    name="test"
    type="radio"
    value=""
  />
  Test 
  <a>
    Give me
  </a>
</label>
`;

exports[`Radio component Should use inline styles 1`] = `
<label
  className="radio"
  disabled={false}
  style={
    Object {
      "height": 250,
    }
  }
>
  <input
    checked={false}
    disabled={false}
    name="test"
    type="radio"
    value=""
  />
  <p>
    Default
  </p>
</label>
`;
